scripts/clean_osm_data.py

```diff
@@ -279,279 +288,320 @@ def finalize_lines_type(df_lines):
-                df.loc[i, list_col[0]] = d[0][0]  # first split  [0]
-                df.loc[i, list_col[1]] = d[1][0]
-                r[list_col[0]] = d[0][1]  # second split [1]
-                r[list_col[1]] = d[1][1]
-                row_list.append(r)
-
-    if row_list:
-        df = pd.concat([df, gpd.GeoDataFrame(row_list, crs=df.crs)], ignore_index=True)
-
-    # if some columns still contain ";" then sum the values
-    for cl_name in list_col:
-        if df[cl_name].dtype == "string":
-            sel_ids = df[cl_name].str.contains(";") == True
-            # TODO: split multiple cell need fix
-            df = df.drop(
-                df.loc[sel_ids, cl_name].index,
-            )
-    return df  # return new frame
-
-
-# cable and circuit tags may be in a non-numeric format
-cables_tag_to_n_cables = {
-    "single": "1",
-    "triple": "3",
-    "partial": "1",
-    "1 disused": "0",
-    "3 disused": "0",
-    "1 (Looped - Haul & Return) + 1 power wire": "1",
-    "ground": "0",
-    "line": "1",
-    # assuming that in case of a typo there is at least one line
-    "`": "1",
-    "^1": "1",
-    "e": "1",
-    "d": "1",
-    "2-1": "3",
-    "3+3": "6",
-    "6+1": "6",
-    "2x3": "6",
-    "3x2": "6",
-    "2x2": "4",
-}
-
-circuits_tag_to_n_circuits = {
-    "1/3": "0",
-    "2/3": "0",
-    # assumption of two lines and one grounding wire
-    "2-1": "2",
-    "single": "1",
-    "partial": "1",
-    "1 disused": "0",
-    # assuming that in case of a typo there is at least one line
-    "`": "1",
-    "^1": "1",
-    "e": "1",
-    "d": "1",
-    "1.": "1",
-}
-
-dropped_cables_tags = [
-    x for x in cables_tag_to_n_cables.keys() if cables_tag_to_n_cables[x] == "0"
-]
-dropped_circuits_tags = [
-    x for x in circuits_tag_to_n_circuits.keys() if circuits_tag_to_n_circuits[x] == "0"
-]
+    # replace raw values
+    repl_freq = {
+        "16.67": "16.7",
+        "50;50;16.716.7": "50;50;16.7;16.7",
+        "50;50;16.716.7": "50;50;16.7;16.7",
+        "50;16.7?": "50;16.7",
+        # "24 kHz": "24000",
+    }
 
+    # TODO: default frequency may be by country
+    # default frequency
+    default_frequency = "50"
 
-def integrate_lines_df(df_all_lines, distance_crs):
+    df["tag_frequency"] = (
+        df["tag_frequency"].replace(repl_freq).fillna(default_frequency)
+    )
+
+    return df
+
+
+def parse_voltage(df):
+    "Function to parse raw voltage column: manual fixing and drop undesired values"
+
+    # replace raw values
+    repl_voltage = {
+        "medium": "33000",
+        "19.1 kV": "19100",
+        "high": "220000",
+        "240 VAC": "240",
+        "2*220000": "220000;220000",
+        "KV30": "30kV",
+    }
+
+    df["voltage"] = (
+        df["voltage"]
+        .replace(repl_voltage)
+        .str.lower()
+        .str.replace(" ", "")
+        .str.replace("_", "")
+        .str.replace("kv", "000")
+        .str.replace("v", "")
+        # .str.replace("/", ";")
```

Comment:
Theoretically yes, but I'd like to keep it to point out that some lines in the dataset are actually separated by / instead of ;.
However, for some entries it works ok but others do not.
I add a comment here